util/patch/patch_test.go

```diff
@@ -79,6 +81,48 @@ func TestHelperUnstructuredPatch(t *testing.T) {
 	g.Expect(afterObj.GetOwnerReferences()).To(Equal(refs))
 }
 
+var _ = Describe("Patch Helper", func() {
+	It("Should patch a corev1.Node object conditions", func() {
```

Comment:
Is it possible to have a test on patching conditions handling concurrency?
e.g.
Create an object 
Get the object and its patcher
Modify the object conditions (without storing it)
Modify the object conditions using another client
Patch the object and see both changes are preserved

Reply:
All the tests needs to be rewritten using envtest. We weren't testing enough before and most of the time I found myself wondering why something was passing